### PR TITLE
Townhall Achievements added, Fisher/Farmer Achievements fixed

### LINKED ISSUE #111: Farmer achievements
### Minecolonies version
v714

I suggest we add them real quick as a hotfix.

### LINKED ISSUE #142: Add Townhall build achievements
We should have an achievement for completing lvl1 and 3 of the Townhall like we have for all other buildings.

### PR DESCRIPTION
**Pull request by nojo121**
Closes #142 
Closes #111 

# Changes proposed in this pull request:
-Added Townhall Achievements
-Fixed Farmer Achievements
-Fixed Fisher Achievements

Review please @Minecolonies/maintainers

### CODE REVIEW

File: src/main/java/com/minecolonies/colony/buildings/BuildingTownHall.java [hunk 1]
```diff
@@ -41,4 +42,13 @@ public View(ColonyView c, BlockPos l)
             return new WindowTownHall(this);
         }
     }
+    public void onUpgradeComplete(final int newLevel)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/colony/buildings/BuildingTownHall.java]: MAJOR Add the "@Override" annotation above this method signature [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1161)
MINOR Document this public method. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: src/main/java/com/minecolonies/achievements/ModAchievements.java [hunk 2]
```diff
@@ -49,6 +49,12 @@
      */
     public static final Achievement achievementBuildingTownhall = new MineColoniesAchievement("townhall", -2, 0, ModBlocks.blockHutTownHall, null)
                                                                     .registerStat();
+    /**
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/achievements/ModAchievements.java]: MINOR First sentence should end with a period. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 3 issues
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/main/java/com/minecolonies/achievements/ModAchievements.java [hunk 3]
```diff
@@ -47,14 +47,27 @@
...
+    public static final Achievement achievementTownhall = new MineColoniesAchievement("townhall", -2, 0, ModBlocks.blockHutTownHall, null)
+
                                                                     .registerStat();
+    /**
+     * Upgrade a townhall to lv 1.
+     */
+    public static final Achievement achievementBuildingTownhall = new MineColoniesAchievement("upgrade.townhall.first", 0, 0, ModBlocks.blockHutTownHall,
+                                                                                                  achievementTownhall).registerStat();
+
+    /**
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/achievements/ModAchievements.java]: MINOR First sentence should end with a period. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 1 issue
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by Kostronor** [comment]
:shipit: @Raycoms @cltnschlosser

**Comment by Raycoms** [comment]
:shipit:

File: src/main/java/com/minecolonies/achievements/ModAchievements.java [hunk 4]
```diff
@@ -47,14 +47,27 @@
     /**
      * Place a townhall.
      */
-    public static final Achievement achievementBuildingTownhall = new MineColoniesAchievement("townhall", -2, 0, ModBlocks.blockHutTownHall, null)
+    public static final Achievement achievementTownhall = new MineColoniesAchievement("townhall", -2, 0, ModBlocks.blockHutTownHall, null)
+
```
Comment (cltnschlosser) [inline src/main/java/com/minecolonies/achievements/ModAchievements.java]: Can we get rid of this newline between the object declaration and the registerStat call

File: src/main/resources/assets/minecolonies/lang/en_US.lang [hunk 5]
```diff
@@ -124,6 +124,10 @@ com.minecolonies.gui.townHall.recall=Recall Citizens
 item.scepterSteel.point=Point one saved
 achievement.upgrade.builder.max=The Builder!
 achievement.townhall.desc=Create your own Settlement
+achievement.upgrade.townhall.first=We need a townhall!
```
Comment (cltnschlosser) [inline src/main/resources/assets/minecolonies/lang/en_US.lang]: @Raycoms @Kostronor Did these strings get added to POE? or Do we add them on merge?

File: src/main/java/com/minecolonies/achievements/ModAchievements.java (hunk 4 above)
Comment (nojo121) [inline src/main/java/com/minecolonies/achievements/ModAchievements.java]: But we've got 3 Achievements.
Place Townhall
upgrade to lv 1
upgrade max

File: src/main/resources/assets/minecolonies/lang/en_US.lang (hunk 5 above)
Comment (nojo121) [inline src/main/resources/assets/minecolonies/lang/en_US.lang]: they were already added to POE.

**Comment by Kostronor** [comment]
:+1: @Raycoms

**Comment by Raycoms** [comment]
:shipit:

**Pull request merged**